src/main/scala/clump/ClumpFetcher.scala

```diff
@@ -35,6 +39,20 @@ final class ClumpFetcher[T, U](source: ClumpSource[T, U]) {
     results
   }
 
+  private def retryIfNeccessary(maybeFailed: Future[Map[T, U]], retries: Int) = {
```

Comment:
I didn't have time to implement this recursion properly obviously :P but you can probably see what I am trying to do